spark/v3.1/build.gradle

```diff
@@ -131,20 +131,21 @@ project(":iceberg-spark:iceberg-spark-extensions-3.1_2.12") {
   }
 
   dependencies {
-    compileOnly "org.scala-lang:scala-library"
+    compileOnly "org.scala-lang:scala-library:${scalaVersion}"
     compileOnly project(path: ':iceberg-bundled-guava', configuration: 'shadow')
     compileOnly project(':iceberg-api')
     compileOnly project(':iceberg-core')
     compileOnly project(':iceberg-data')
     compileOnly project(':iceberg-orc')
     compileOnly project(':iceberg-common')
     compileOnly project(':iceberg-spark:iceberg-spark-3.1_2.12')
-    compileOnly("org.apache.spark:spark-hive_2.12:${sparkVersion}") {
+    compileOnly("org.apache.spark:spark-hive_${scalaVersion}:${libs.versions.sparkHive31.get()}") {
       exclude group: 'org.apache.avro', module: 'avro'
       exclude group: 'org.apache.arrow'
       exclude group: 'org.apache.parquet'
       exclude group: 'org.roaringbitmap'
     }
+    compileOnly platform(libs.jackson211.bom)
```

Comment:
why do we need to add this dependency now?

Reply:
removed